x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -110,48 +70,22 @@ protected DataType loadTypeFromName(String typeName) {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(fieldName);
-        if (out.getVersion().onOrAfter(SWITCHED_FROM_DOCVALUES_TO_SOURCE_EXTRACTION)) {
-            out.writeOptionalString(fullFieldName);
-        }
         out.writeOptionalString(dataType == null ? null : dataType.typeName());
-        out.writeBoolean(useDocValue);
         out.writeOptionalString(hitName);
         out.writeBoolean(arrayLeniency);
     }
 
     @Override
     public Object extract(SearchHit hit) {
         Object value = null;
-        if (useDocValue) {
-            DocumentField field = hit.field(fieldName);
-            if (field != null) {
-                value = unwrapMultiValue(field.getValues());
-            }
-        } else {
-            // if the field was ignored because it was malformed and ignore_malformed was turned on
-            if (fullFieldName != null
-                    && hit.getFields().containsKey(IgnoredFieldMapper.NAME)
-                    && isFromDocValuesOnly(dataType) == false) {
-                /*
-                 * We check here the presence of the field name (fullFieldName including the parent name) in the list
-                 * of _ignored fields (due to malformed data, which was ignored).
-                 * For example, in the case of a malformed number, a "byte" field with "ignore_malformed: true"
-                 * with a "text" sub-field should return "null" for the "byte" parent field and the actual malformed
-                 * data for the "text" sub-field.
-                 */
-                if (hit.getFields().get(IgnoredFieldMapper.NAME).getValues().contains(fullFieldName)) {
-                    return null;
-                }
-            }
-            Map<String, Object> source = hit.getSourceAsMap();
-            if (source != null) {
-                value = extractFromSource(source);
-            }
+        DocumentField field = hit.field(fieldName);
+        if (field != null) {
+            value = unwrapFieldsMultiValue(field.getValues());
         }
         return value;
     }
 
-    protected Object unwrapMultiValue(Object values) {
+    protected Object unwrapFieldsMultiValue(Object values) {
```

Comment:
Why the "Field**s**" plural?

Reply:
Good question. If I remember well, it's a leftover from a previous attempt where both strategies were used at the same time in code: extraction from source and docvalues (`unwrapSourceMultiValue`) and extraction from "fields" API output (`unwrapFieldsMultiValue`). No need for the rename, indeed. I'll revert.